### PR TITLE
FIT-40 Implement test bubble execution on build

### PR DESCRIPTION
**Pull request by brunoOSI**
### Jira Story

- [FIT-40 Implement test bubble execution on build](https://osi-cwds.atlassian.net/browse/FIT-40)

This makes use of : [FIT-38 Create base images for Intake containers to speed up builds](https://osi-cwds.atlassian.net/browse/FIT-38)

Also rubyfied parts of the Makefile (an ongoing effort that will be continued over this sprint)

## Tests
- [ ] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [x] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: scripts/ci/compute_version.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+base_version = `git describe --tags $(git rev-list --tags --max-count=1)`.chomp
```
Comment (NagaPradeepGajula-osi) [inline scripts/ci/compute_version.rb]: So, this will still keep the version tag instead of `latest` when we deploy to preint in automated fashion right?

Comment (brunoOSI) [reply scripts/ci/compute_version.rb]: No, this just computes the tag version. Your comment made me check the Jenkinsfile once more and I noticed the `make tag latest <version>`  was missing!
Thanks for that. I'm fixing the Jenkinsfile.

Comment (NagaPradeepGajula-osi) [reply scripts/ci/compute_version.rb]: 👍

Comment (brunoOSI) [reply scripts/ci/compute_version.rb]: Converted the publishing part to ruby.

**Comment by brunoOSI** [comment]
retest this please

File: Jenkinsfile [hunk 2]
```diff
@@ -72,40 +81,36 @@ node('intake-slave') {
...
         currentBuild.result = 'FAILURE'
         throw e
     }
+
     finally {
         try {
-          if (branch == 'master') {
-             //Send slack alert only if it is master
+            stage('Clean') {
+                retry(2) {
```
Comment (TylerOSI) [inline Jenkinsfile]: why are we re-trying `make clean` ?

File: docker/base_images/Dockerfile.intake_testing [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    libfreetype6-dev \
+    libfontconfig1 \
+    libfontconfig1-dev \
+    yarn \
+    bzip2
+
+RUN curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
+RUN apt-get install -y nodejs
+
+ADD https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2 /tmp
```
Comment (TylerOSI) [inline docker/base_images/Dockerfile.intake_testing]: I assume this is a precompiled binary. Is there no yummy package we could install with `apt-get`  (then we wouldn't need an install script)?

File: docker/base_images/Dockerfile.intake_testing [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ADD ./bin/install_phantomjs /tmp/install_phantomjs
+RUN /tmp/install_phantomjs
+ARG GECKODRIVER_VERSION=0.18.0
+RUN wget --no-verbose -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GECKODRIVER_VERSION/geckodriver-v$GECKODRIVER_VERSION-linux64.tar.gz \
+  && rm -rf /opt/geckodriver \
+  && tar -C /opt -zxf /tmp/geckodriver.tar.gz \
+  && rm /tmp/geckodriver.tar.gz \
+  && mv /opt/geckodriver /opt/geckodriver-$GECKODRIVER_VERSION \
+  && chmod 755 /opt/geckodriver-$GECKODRIVER_VERSION \
+  && ln -fs /opt/geckodriver-$GECKODRIVER_VERSION /usr/bin/geckodriver
```
Comment (TylerOSI) [inline docker/base_images/Dockerfile.intake_testing]: if there is any yummy package for gecko-driver that we could install with `apt-get` at a specific version -- that would slim this down a little.

emphasis on 'if'

File: docker/base_images/Dockerfile.intake_testing (hunk 3 above)
Comment (brunoOSI) [reply docker/base_images/Dockerfile.intake_testing]: There's no debian package nor PPS source for this, no.

File: docker/base_images/Dockerfile.intake_testing (hunk 4 above)
Comment (brunoOSI) [reply docker/base_images/Dockerfile.intake_testing]: It seems to be a PPA for geckodriver from Mozilla, but I might change that in a different story. 
It involves debugging the test suite and that would be too much for this story.

**Comment by brunoOSI** [comment]
The clean stage is failing on PRs, I'm adding another cleanup stage more flexible.

File: scripts/ci/clean.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  puts '==> Removing release artifacts'
+  `rm -rf release`
+  `docker-compose #{release_args} down --volumes --remove-orphans --rmi all`
+end
+
+puts '==> Removing test artifacts'
+`docker-compose #{test_args} down --volumes --remove-orphans --rmi all`
+
+puts '==> Removing generated images'
+docker_images = `docker images -q -f label=application=intake_accelerator`
```
Comment (TylerOSI) [inline scripts/ci/clean.rb]: `label=application=intake_accelerator` is a little odd syntax -- is that intentional?

File: Jenkinsfile [hunk 6]
```diff
@@ -91,9 +91,7 @@ node('intake-slave') {
     finally {
         try {
             stage('Clean') {
-                retry(2) {
-                    sh 'make clean'
-                }
+                sh './scripts/ci/clean.rb'
```
Comment (TylerOSI) [inline Jenkinsfile]: 💯

**Pull request merged**